### PR TITLE
Bellnc code review 1

### PR DESCRIPTION
**Pull request by alphapixel1**
- Changed all DAO's to interfaces
- flushed out settings dao
- created settings dto
- added properties to character and player class to match the readme

### CODE REVIEW

**Comment by alphapixel1** [comment]
## Analysis of the program.
Program is extremely behind, with the final project due in 30 days, a lot of work will need to be done. Like everything.

## Was the program available in UC Github on time?
Yes? I don't know what you mean by  UC Github

## Is the program documented/commented well enough for you to understand?
There are no comments and no real code in the project that wasn't generated by android studio. 

## Does the program compile?
Yes

## Rationale behind your changes.
The program was bare-bones when I got to it my changes were an attempt to prepare the DAOs and DTOs for use, I did this by changing all of the DAOs to interfaces, added some queries to the settings DAO and flushed out player and character dtos and created a dto for settings and flushed it out in accordance with the readme.

## Technical Concepts Learned
I learned nothing from reviewing this code.

## 3 commits made to my groups github repository before the end of sprint deadline
- https://github.com/alphapixel1/Keybind-Helper-Android/commit/9a791c5550004c71c53ab1b77dc1834aaa88d206
- https://github.com/alphapixel1/Keybind-Helper-Android/commit/32cbfe1f027eb674c9060345f65fcba162be370f
- https://github.com/alphapixel1/Keybind-Helper-Android/commit/233e328bb165808089f9dba629edbf94b2709eea
### Note: my group met together in person and worked to develop these repos:
https://github.com/alphapixel1/KeybindHelperV2
https://github.com/alphapixel1/Android-Keybind-Helper-V3
These repos later became branch: [Keybind-helper-v3-imported-into-keybind-helper-v1](https://github.com/alphapixel1/Keybind-Helper-Android/tree/Keybind-helper-v3-imported-into-keybind-helper-v1),
The most up to date and functioning branch is [Sprint-Code-Review-1 ](https://github.com/alphapixel1/Keybind-Helper-Android/tree/Sprint-Code-Review-1).
Master is currently broken due to a merge issue, I converted every file to kotlin but forgot to remove the original files in main so now its a mess.
I am not really sure when the sprint review deadline was but just incase those 3 were not sufficient, here is a screenshot of my work on the master branch
image

File: app/src/main/java/com/animal_villa/dao/ICharacterListDAO.kt [hunk 1]
```diff
@@ -1,4 +1,7 @@
 package com.animal_villa.dao
 
-class ICharacterListDAO {
+import androidx.room.Dao
+
+@Dao
+interface ICharacterListDAO {
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dao/ICharacterListDAO.kt:6]: This was recommended in a previous code review.  When that happens, only first in wins.

File: app/src/main/java/com/animal_villa/dao/IPlayAreaDAO.kt [hunk 2]
```diff
@@ -1,4 +1,7 @@
 package com.animal_villa.dao
 
-class IPlayAreaDAO {
+import androidx.room.Dao
+
+@Dao
+interface IPlayAreaDAO {
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dao/IPlayAreaDAO.kt:6]: Ditto note above.

File: app/src/main/java/com/animal_villa/dto/Animal.kt [hunk 3]
```diff
@@ -1,4 +1,7 @@
 package com.animal_villa.dto
 
+import androidx.room.Entity
+
+@Entity
 class Animal {
 }
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dto/Animal.kt:7]: This isn't a data class, and there aren't any fields here to persist, so @Entity does not make sense in this context.

File: app/src/main/java/com/animal_villa/dto/Character.kt [hunk 4]
```diff
@@ -1,4 +1,11 @@
...
+import androidx.room.Entity
 
+@Entity
 class Character {
+    lateinit var name:String
+    var age:Int=0;
+    lateinit var job:String
+    lateinit var statusValues:MutableList<Int> //TODO dont know what a status value is so I made it an int
+
 }
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dto/Character.kt:11]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/animal_villa/dto/Player.kt [hunk 5]
```diff
@@ -1,4 +1,11 @@
 package com.animal_villa.dto
 
+import androidx.room.Entity
+
+@Entity
 class Player {
+    var Status:Int=100;
+    var Money=0;
+    var Energy=30;
+    var CurrentLocation=0;
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dto/Player.kt:10]: variable names should be lowerCamelCase in Kotlin.

This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/animal_villa/dto/Settings.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.animal_villa.dto
+
+import androidx.room.Entity
+import com.animal_villa.dao.ISettingsDAO
+
+@Entity
+class Settings {
+    var volume = 1
+    var textSpeed = 1
+    var soundEffects = true
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dto/Settings.kt:10]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/animal_villa/dto/Settings.kt [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.animal_villa.dto
+
+import androidx.room.Entity
+import com.animal_villa.dao.ISettingsDAO
+
+@Entity
+class Settings {
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dto/Settings.kt:7]: Should this be a data class?

File: app/src/main/java/com/animal_villa/dto/Settings.kt [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    var volume = 1
+    var textSpeed = 1
+    var soundEffects = true
+    fun updateSettings(dao: ISettingsDAO) {
+        dao.updateSettings(this)
+    }
```
Comment (discospiff) [inline app/src/main/java/com/animal_villa/dto/Settings.kt:13]: A data class/DTO/noun class should not invoke a DAO to save itself. Data class/noun class/DTO should only have attributes.  The business logic, persistence logic, etc., should be handled by a different class.

**Comment by discospiff** [comment]
Additions are likely needed for the project at some point, but are new, and sometimes incomplete, functionality.  New functionality is better in a separate branch until fully developed, as features should not be merged to main/master until the meet the team's Definition of Done.  Code review branches should minimize technical debt, and thus, be merge-able to master immediately.  So, if the team desires these features, I'd keep the branch open, and iterate over the feature until it is complete.

**Comment by portekn** [comment]
Some files are no longer in use or have been renamed

**Pull request closed without merging**